agent/agents/postgres/pgstatstatements/pgstatstatements.go

```diff
@@ -91,11 +94,20 @@ func New(params *Params, l *logrus.Entry) (*PGStatStatementsQAN, error) {
 	reformL := sqlmetrics.NewReform("postgres", params.AgentID, l.Tracef)
 	// TODO register reformL metrics https://jira.percona.com/browse/PMM-4087
 	q := reform.NewDB(sqlDB, postgresql.Dialect, reformL).WithTag(queryTag)
-	return newPgStatStatementsQAN(q, sqlDB, params.AgentID, params.MaxQueryLength, params.DisableCommentsParsing, l)
+	// read pg_stat_statements.max parameter
+	var cacheSize uint
+	err = q.QueryRow(pgssMaxQuery).Scan(&cacheSize)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get pg_stat_statements.max")
+	}
+	if cacheSize < minPgSSCacheSize {
+		cacheSize = minPgSSCacheSize
+	}
+	return newPgStatStatementsQAN(q, sqlDB, params.AgentID, cacheSize, params.MaxQueryLength, params.DisableCommentsParsing, l)
```

Comment:
let's move it inside `newPgStatStatementsQAN` so it will be consistent with MySQL perfschema codebase.